File: frontend/lib/core/models/extra_meal.dart
class ExtraMeal {
  final String id;
  final String userId;
  final DateTime date;
  final String mealType;
  final String? note;
  final String createdBy;
  final DateTime createdAt;

  const ExtraMeal({
    required this.id,
    required this.userId,
    required this.date,
    required this.mealType,
    this.note,
    required this.createdBy,
    required this.createdAt,
  });

  factory ExtraMeal.fromJson(Map<String, dynamic> json) => ExtraMeal(
        id: json['id'] as String,
        userId: json['user_id'] as String,
        date: DateTime.parse(json['date'] as String),
        mealType: json['meal_type'] as String,
        note: json['note'] as String?,
        createdBy: json['created_by'] as String,
        createdAt: DateTime.parse(json['created_at'] as String),
      );
}

File: backend/app/schemas/extra_meal.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.meal_skip import MealType


class ExtraMealCreate(BaseModel):
    user_id: uuid.UUID
    date: date
    meal_type: MealType
    note: str | None = Field(default=None, max_length=255)


class ExtraMealResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    date: date
    meal_type: MealType
    note: str | None
    created_by: uuid.UUID
    created_at: datetime
